hat/assets/js/apps/Iaso/domains/dataSources/requests.js

```diff
@@ -317,6 +317,19 @@ export const useCopyDataSourceVersion = () => {
     );
 };
 
+export const useCreateSourceVersion = () => {
+    return useSnackMutation(
+        ({ description, dataSourceId }) => {
+            return postRequest(`/api/sourceversions/`, {
+                description,
+                data_source_id: dataSourceId,
+            });
+        },
+        MESSAGES.newEmptyVersionSavedSuccess,
+        MESSAGES.newEmptyVersionError,
+    );
+};
+
```

Comment:
Please define the function outside of the hook and pass an object as parameter to `useSnackMutation`, like:

```
// ...
return useSnackMutation({
    mutationFn: saveSourceVersion, // the function you moved out of your hook
    snackErrorMsg: MESSAGES.newEmptyVersionError,
    snackSuccessMessage: MESSAGES.newEmptyVersionSavedSuccess,
    invalidateQueryKey: 'dataSourceVersions' // this make sure new versions are downloaded
});
```

The hooks in that part of the code are quite old, so it's natural that you replicated their structure, but since we can now pass objects to both `useSnackQuery` and `useSnackMutation`, it's better do it this way as it is easier to read